HandleNHNotificationCallActivity/handler.ts

```diff
@@ -53,9 +55,20 @@ export const getCallNHServiceActivityHandler = (
       context.log.info(
         `${logPrefix}|${message.kind}|INSTALLATION_ID=${message.installationId}`
       );
+
+      const nhService = getNHLegacyService();
+
+      if (isNone(nhService)) {
```

Comment:
We should try to avoid an imperative form as much as possible. One way of seeing this is to consider each step of a chain as a manipulation of our data type into a new data type. 

For example, we could have written a step like
```ts
.chain(({ message }) =>
  // Either to TaskEither
  fromEither( 
    // Option to Either
    fromOption( 
      // this is the "left" of the Either in case the option is none
      new Error("Cannot determine legacy Notification Hub instance"),
     // this is the actual option
      getNHService(message.installationId)
    )
  )
  // this is needed to pass the message through
  .map(nhService => ({ message, nhService })  
)
.chain(({ message, nhService }) => {
....
```

Benefit: you only rely on well-tested, reusable bits with known semantic; this increases standardisation of the code and makes core review easier, as well as leverage the type system to check our flow control logic.

Reply:
@balanza , if I remove the "Option" from the "getLegacyNH" method, we don't anymore need this step. 
But Thanks for the explanation, it's a little bit tricky at the beginning.